lib/types.nix

```
@@ -548,17 +549,23 @@ rec {
       name = "listOf";
       description = "list of ${optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType}";
       descriptionClass = "composite";
-      check = isList;
+      check = x: isList x || x._type or "" == "forAllValues";
       merge = loc: defs:
+        let
+          partitionedDefs = partition (d: d.value._type or "" == "forAllValues") defs;
+          forAllValueses = partitionedDefs.right;
+          defs' = partitionedDefs.wrong;
```

By inlining single-use bindings like these, we could get rid of a few `Env` entries.

To avoid losing the documentation-like role of the names, those could be documented in a comment on `partitionedDefs` instead.

```suggestion
          # partitionedDefs.right: all forAllValues definitions
          # partitionedDefs.wrong: all regular definitions
          partitionedDefs = partition (d: d.value._type or "" == "forAllValues") defs;
```

---

Same in the other types.